src/ui/components/shared/LoginModal.tsx

```diff
@@ -25,7 +19,7 @@ function LoginModal() {
           <div className="flex flex-col items-center">
             <button
               type="button"
-              onClick={onClick}
+              onClick={() => loginAndReturn()}
```

Comment:
Can just do `onClick={loginAndReturn}` here, no?

Reply:
We could. I tend to avoid passing non-event handlers to event callbacks so things don't accidentally break if we change the signature of `loginAndReturn`. TS will probably catch that for us but it's an easy regression to introduce.